src/gluonts/dataset/split.py

```diff
@@ -73,130 +73,87 @@
 
 from abc import ABC, abstractmethod
 from dataclasses import dataclass
-from typing import cast, Generator, List, Optional, Tuple
+from typing import Generator, Optional, Tuple
 
-import numpy as np
 import pandas as pd
-import pydantic
 
 from gluonts.dataset import Dataset, DataEntry
 from gluonts.dataset.field_names import FieldName
 
 
-class TimeSeriesSlice(pydantic.BaseModel):
-    """
-    Like DataEntry, but all time-related fields are of type pd.Series and is
-    indexable, e.g `ts_slice['2018':]`.
-    """
-
-    class Config:
-        arbitrary_types_allowed = True
-
-    target: pd.Series
-    item_id: Optional[str] = None
+def slice_data_entry(
+    entry: DataEntry, slc: slice, prediction_length: int = 0
+) -> DataEntry:
+    if slc.stop is not None:
+        slc_extended = slice(slc.start, slc.stop + prediction_length, slc.step)
```

Comment:
The sum here could have different sign compared to `slc.stop` (if `slc.stop` is negative); need to make this more robust